bounty_hunter/src/main.rs

```diff
@@ -1,139 +1,124 @@
 use dotenv::dotenv;
+use env_logger::Builder;
+use log::LevelFilter;
+use openssl::ssl::{SslAcceptor, SslFiletype, SslMethod};
 
-use std::env;
-use std::io::Read;
-use std::sync::Mutex;
-
-pub mod models;
-pub mod schema;
-use self::models::*;
+use std::{env, path, sync::Mutex};
 
-use self::schema::status::dsl::*;
+use network_endpoints::{handle_get_channel_state, handle_upload_channel_state};
 
-pub fn establish_connection() -> SqliteConnection {
-    dotenv().ok();
+lazy_static! {
+    static ref DB_CONN: Mutex<SqliteConnection> = {
+        dotenv().ok();
 
-    let database_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");
-    SqliteConnection::establish(&database_url)
-        .expect(&format!("Error connecting to {}", database_url))
+        let database_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");
+        let conn = SqliteConnection::establish(&database_url)
+            .expect(&format!("Error connecting to {}", database_url));
+        Mutex::new(conn)
+    };
 }
 
-#[derive(Serialize, Deserialize, Debug)]
-pub struct BountyUpdate {
-    pub from: Identity,
-    pub balance: Int256,
-    pub tx: PaymentTx,
-}
+static BOUNTY_HUNTER_PORT: u16 = 4878u16;
 
 fn main() {
-    env_logger::init();
-    trace!("Starting");
-
-    let conn = Mutex::new(establish_connection());
-
-    rouille::start_server("[::0]:8888", move |request| {
-        // TODO: fix the port
-        router!(request,
-            (POST) (/update) => {
-                process_updates(request, &conn)
-            },
-            (GET) (/list) => {
-                list_status(request, &conn)
-            },
-            _ => rouille::Response::empty_404()
-        )
-    });
-}
-
-fn process_updates(request: &Request, conn: &Mutex<SqliteConnection>) -> Response {
-    let conn = conn.lock().unwrap();
-    if let Some(mut data) = request.data() {
-        let mut status_str = String::new();
-        data.read_to_string(&mut status_str).unwrap();
-        let update: BountyUpdate = serde_json::from_str(&status_str).unwrap();
-        trace!("Received update, status: {:?}", update);
-        trace!("Received update, balance: {}", update.balance);
+    match env::var("RUST_LOG") {
+        Ok(_) => env_logger::init(),
+        Err(_) => Builder::new().filter_level(LevelFilter::Info).init(),
+    }
 
-        let stat = Status {
-            ip: String::from(format!("{}", update.from.mesh_ip)),
-            mac: String::from(format!("{}", update.from.wg_public_key)),
-            balance: String::from(format!("{}", update.balance)),
-        };
+    info!("Althea Bounty Hunter {}", env!("CARGO_PKG_VERSION"));
 
-        trace!("Checking if record exists for {}", stat.ip);
+    let system = actix::System::new("main");
 
-        let exists = select(exists(status.filter(ip.eq(stat.ip.clone()))))
-            .get_result(&*conn)
-            .expect("Error loading statuses");
+    // Used with or without HTTPS
```

Comment:
I know on the routers we really hate to panic, but this is server code, important server code at that, it simply shouldn't start without https because we rely on it to function properly.

Reply:
Done, HTTPS only, bailing otherwise.